cmd/operator-sdk/run/cmd.go

```
@@ -15,21 +15,95 @@
+	hoflags "github.com/operator-framework/operator-sdk/pkg/helm/flags"
+
+	log "github.com/sirupsen/logrus"
 	"github.com/spf13/cobra"
 )
 
-// NewCmd returns a command that contains subcommands to run specific
-// operator types.
+type runCmd struct {
+	// Common options.
+	kubeconfig string
+	namespace  string
+
+	// Run type.
+	olm, local bool
+
+	// Run type-specific options.
+	olmArgs   olmoperator.OLMCmd
+	localArgs runLocalArgs
+}
+
 func NewCmd() *cobra.Command {
-	runCmd := &cobra.Command{
+	c := &runCmd{}
+	cmd := &cobra.Command{
 		Use:   "run",
-		Short: "Runs a generic operator",
-		Long: `Runs a generic operator. This is intended to be used when running
-in a Pod inside a cluster. Developers wanting to run their operator locally
-should use "up local" instead.`,
+		Short: "Run an Operator in a variety of environments",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			projutil.MustInProjectRoot()
+
+			switch {
+			case c.olm:
+				c.olmArgs.KubeconfigPath = c.kubeconfig
+				c.olmArgs.OperatorNamespace = c.namespace
+				if c.olmArgs.ManifestsDir == "" {
+					operatorName := filepath.Base(projutil.MustGetwd())
+					c.olmArgs.ManifestsDir = filepath.Join(olmcatalog.OLMCatalogDir, operatorName)
+				}
+				if err := c.olmArgs.Run(); err != nil {
+					log.Fatalf("Failed to run operator using OLM: %v", err)
+				}
+			case c.local:
+				// Get default namespace to watch if unset.
+				if !cmd.Flags().Changed("namespace") {
+					_, defaultNamespace, err := k8sinternal.GetKubeconfigAndNamespace(c.kubeconfig)
+					if err != nil {
+						return fmt.Errorf("error getting kubeconfig and default namespace: %v", err)
+					}
+					c.namespace = defaultNamespace
+				}
+				c.localArgs.kubeconfig = c.kubeconfig
+				c.localArgs.namespace = c.namespace
+				if err := c.localArgs.run(); err != nil {
+					log.Fatalf("Failed to run operator locally: %v", err)
+				}
+			}
+			return nil
```

Should return an error if both `c.olm` and `c.local` are false?